modules/renter/hostdb/hostentry.go

```diff
@@ -0,0 +1,115 @@
+package hostdb
+
+import (
+	"math"
+
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+// updateHostDBEntry updates a HostDBEntries's historic interactions if more
+// than one block passed since the last update. This should be called every time
+// before the recent interactions are updated.  if passedTime is e.g. 10, this
+// means that the recent interactions were updated 10 blocks ago but never
+// since. So we need to apply the decay of 1 block before we append the recent
+// interactions from 10 blocks ago and then apply the decay of 9 more blocks in
+// which the recent interactions have been 0
+func updateHostHistoricInteractions(host *modules.HostDBEntry, bh types.BlockHeight) {
+	passedTime := bh - host.LastHistoricUpdate
+	if passedTime == 0 {
+		// no time passed. nothing to do.
```

Comment:
hmm. If the consensus db reset or something, this could actually cause issues. Good catch